### ISSUE TITLE
Training Code

### ISSUE THREAD

**Issue by Onehour0108** [Xiyue-Wang/1st-in-MICCAI-MIDOG-2021-challenge#3]
你好，感谢你们提供的代码。我们对你们的工作很感兴趣。关于你们的方法，我们有下面的疑问需要你们的帮助：
由于本工程没有提供训练代码，我们按照文章提供的参数和细节搭建了一个训练框架以复现你们的训练过程，细节如下：
Patch size:512*512
Batch size:4
Learning rate:0.0003
Weight decay:0.0001
Adam optimizer
Loss: focalloss+diceloss
我们严格按照论文细节进行训练过程的复现，然而我们训练得到的结果并不正常。
请问训练代码是否还有额外需要注意的细节被我们忽略了，或者能否提供一下你们官方的训练代码帮助我们，期待你的回复。

**Comment by Xiyue-Wang** [comment]
你好，你生成了细胞核的分割了吗？我们的方法都是交给主办方，权重和代码，在域外评测的

**Comment by Onehour0108** [comment]
是的，我们使用了hover-net分割结果得到mask，并且在我们的训练框架上使用简单的fcn网络结构，是可以得到分割结果的，尽管分割结果不太理想。但是，当我们尝试将网络换为FMDet时，得到的分割结果却毫无规律。

**Comment by Xiyue-Wang** [comment]
那不应该，top 1-2的队伍都是这样干的，你是效果很差吗？我看看你的代码的细节？

**Comment by Onehour0108** [comment]
感谢你的回复，我们的训练框架大致可见https://github.com/Onehour0108/train_FDMet，其中logs里的图片为相应iter的预测结果，分别对应的阈值为0.5，0.94.

**Comment by Xiyue-Wang** [comment]
我看了哈你的代码，你的验证集上的F1分数是多少？你还有个可能是，你的数据没有采样，你可以看我们论文，因为正负样本不均衡，需要一个batch里均衡采样

**Comment by Onehour0108** [comment]
我们在验证集上预测都是乱的，f1极低接近于0，我的猜想是，我的哪个设置是错误的，导致模型根本没学对东西，并不是说模型的性能不够高。我们的训练样本是有平衡的，关于正负样本，我们的理解是，包含有丝分裂的512*512patch为阳性样本，不包含则为阴性，不知道这样理解对不对

**Comment by Onehour0108** [comment]
我们有个疑惑，你们计算loss使用的是模型输出的Batch_size*1*512*512 的特征吗，因为我们观察你们模型与FCN的区别在于，输出的特征维度不同

**Comment by Xiyue-Wang** [comment]
![image](https://user-images.githubusercontent.com/61115068/227137625-7207899f-3789-464b-8b28-def1c4c66952.png)
我看了哈，应该是你loss没算对，你好好check哈你的loss，我们不是你这样算的，没有-1，我们的输出是Batch_size*512*512，

**Comment by Onehour0108** [comment]
> 我们有个疑惑，你们计算loss使用的是模型输出的Batch_size*1*512*512 的特征吗，因为我们观察你们模型与FCN的区别在于，输出的特征维度不同

**Comment by Xiyue-Wang** [comment]
你loss 现在正常了吗

**Comment by Onehour0108** [comment]
我们重新更换了loss，但是似乎还是不太行。我们尝试直接使用你们的权重，并且已经提出了申请

**Comment by Pptqlgc** [comment]
> 我们重新更换了loss，但是似乎还是不太行。我们尝试直接使用你们的权重，并且已经提出了申请
请问你们最后问题解决了吗？最后得到的效果好嘛？

**Comment by Pptqlgc** [comment]
请问你们最后问题解决了吗？最后得到的效果好嘛？

**Comment by Pptqlgc** [comment]
请问你们最后问题解决了吗？最后得到的效果怎么样？ @Onehour0108
